src/main/webapp/common_util.js

```diff
@@ -13,16 +13,22 @@
 // limitations under the License.
 
 /*
- * Extracts video id from full url
- */ 
-function cleanseUrl(url) {
-  // Split web address from parameters, extract first parameter
-  // TODO: Add checks to make this work if video is not first parameter.
-  var videoId = url.split("?");
-  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
-
-  // If param name present, remove it to isolate video Id.
-  videoId = videoId.replace("v=", "");
-  
+ * @param url: url of the video to be analyzed
+ * @returns: just the video id of that video
+ * Credit for RegEx method derived from top answer on: 
+ * https://stackoverflow.com/questions/28735459/how-to-validate-youtube-url-in-client-side-in-text-box
+ */	 
+function extractYouTubeUrl(url) {
+  var videoId = '';
+  var idLength = 11; // Length of video Id's in YouTube videos
+  if (url != undefined || url != '') {
+    // This regular expression represents the different patterns that may occur in a YouTube URL
+    var regExp = /^.*(youtu.be\/|v\/|u\/\w\/|e\/|embed\/|watch\?v=|\&v=|\?v=|\?vi=|v\&)([^#\&\?]*).*/;
+    var match = url.match(regExp);
+    // Check for proper format, focus on segment that contains videoId
+    if (match && match[2].length == idLength) {
+      videoId = match[2];
```

Comment:
You can make this function a little simpler by just returning here.

I.e. `return match[2];`.

This will also mean that you can get rid of the `videoId` variable.

Reply:
Makes sense; my only concern is that if for some reason an invalid ID with a length > 11 is entered it will cause an error.